src/sentry/web/frontend/react_page.py

```diff
@@ -95,6 +95,15 @@ def handle_react(
         ):
             user_theme = f"theme-{react_config['user']['options']['theme']}"
 
+        prefers_chonk_ui = False
+        if features.has("organizations:chonk-ui", org_context):
+            if features.has("organizations:chonk-ui-enforce", org_context):
+                prefers_chonk_ui = True
+            elif react_config.get("user", None) and react_config["user"].get("options", {}).get(
+                "prefersChonkUI", False
+            ):
+                prefers_chonk_ui = react_config["user"]["options"]["prefersChonkUI"]
```

Comment:
`chonk-ui` flag is required. uses `chonk-ui-enforce` if present, user preference otherwise